src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -10989,6 +10969,17 @@ static bool isCandidateUnique(ref MethodSymbol? method, MethodSymbol candidate)
             }
         }
 
+        private static bool IsStaticInstanceMismatch(BoundExpression receiver, Symbol extensionMember)
+        {
+            bool memberCountsAsStatic = extensionMember is MethodSymbol { IsExtensionMethod: true } ? false : extensionMember.IsStatic;
```

Comment:
>bool memberCountsAsStatic = extensionMember is MethodSymbol { IsExtensionMethod: true } ? false : extensionMember.IsStatic; [](http://example.com/codeflow?start=12&length=123)

This logic is too special for a method with such "innocent" name. It looks like the method is meant to be used only under specific conditions. Consider renaming to avoid an accidental misuse of the method.